### PR TITLE
Fixed #135 - Added an RPC remix route on publish

### LINKED ISSUE #135: Implement /remix/:id route
This should copy the publishedProject into a project for the authorized user.

### PR DESCRIPTION
**Pull request by sedge**

### CODE REVIEW

File: api/modules/publishedProjects/controller.js [hunk 1]
```diff
@@ -1,5 +1,59 @@
+var Promise = require('bluebird');
+
+var errors = require('../../classes/errors');
 var BaseController = require('../../classes/base_controller');
+var model = require('./model');
```
Comment (gideonthomas) [inline api/modules/publishedProjects/controller.js]: Shouldn't this be somewhere around line 8 with the other models?

File: api/modules/publishedProjects/controller.js [hunk 2]
```diff
@@ -1,5 +1,59 @@
...
+      });
+  }
+
+  function duplicateProject() {
+    return Projects.forge({
+      title: publishedProject.get('title') + ' (remix)',
+      user_id: user.get('id'),
+      tags: publishedProject.get('tags'),
+      description: publishedProject.description,
+      date_created: req.payload.now,
```
Comment (gideonthomas) [inline api/modules/publishedProjects/controller.js]: check if this is actually how you access query params

File: api/modules/publishedProjects/controller.js [hunk 3]
```diff
@@ -1,5 +1,59 @@
...
+        }
+      }).fetchAll();
+    }
+
+    function duplicateFiles(publishedFiles) {
+      return Promise.map(publishedFiles.records, function(publishedFile) {
+        return Files.forge({
+          path: publishedFile.get('path'),
+          project_id: newProject.get('id'),
+          buffer: newProject.get('buffer')
```
Comment (gideonthomas) [inline api/modules/publishedProjects/controller.js]: newProject?

File: api/modules/publishedProjects/routes/create.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+var prereqs = require('../../../classes/prerequisites');
```
Comment (gideonthomas) [inline api/modules/publishedProjects/routes/create.js]: probably rename it to remix.js

**Comment by gideonthomas** [comment]
r+ with nits

File: api/modules/publishedProjects/controller.js (hunk 3 above)
Comment (sedge) [inline api/modules/publishedProjects/controller.js]: Good catch!

**Comment by gideonthomas** [comment]
r+

**Pull request merged**